### PR TITLE
[PSM Interop] Outlier detection: use native cpp server since v1.57.x

### PR DESCRIPTION
**Pull request by sergiitk**
Related:
- CPP implementation: be99673d069217def9036cc9ae0346da324ba373, d55431995c77573c17cae1b78a931aa7b2ecf991.
- Feature request in go: https://github.com/grpc/grpc-go/issues/6288
- Feature request in python: https://github.com/grpc/grpc/issues/33134

FYI @eugeneo @zasweq

### CODE REVIEW

**Comment by sergiitk** [comment]
Test runs:
- [ ] [grpc/core/v1.55.x/branch/linux/grpc_xds_k8s_lb](https://source.cloud.google.com/results/invocations/21b9089a-6b49-456f-9e69-e10ed49edc47)
- [ ] [grpc/core/v1.54.x/branch/linux/grpc_xds_k8s_lb](https://source.cloud.google.com/results/invocations/585f1ea2-7885-4e1a-b40e-7b166439a6be)
- [ ] [grpc/node/v1.8.x/xds_k8s_lb](https://source.cloud.google.com/results/invocations/94d413ad-60e6-4ed7-8940-deb386e52189)

**Comment by sergiitk** [comment]
Tests passed, but the override didn't work.

**Comment by sergiitk** [comment]
New set of tests:
- [ ] [grpc/core/v1.55.x/branch/linux/grpc_xds_k8s_lb](https://source.cloud.google.com/results/invocations/2f19f618-abf7-4fd6-943c-0caddb408768)
- [ ] [grpc/core/v1.54.x/branch/linux/grpc_xds_k8s_lb](https://source.cloud.google.com/results/invocations/a08b332f-6555-46b7-acc4-bf4248ee4685)
- [ ] [grpc/node/v1.8.x/xds_k8s_lb](https://source.cloud.google.com/results/invocations/596ddc93-6912-49c6-ba7e-afad06521f1f)

**Comment by sergiitk** [comment]
Interesting. The cpp server didn't work after all.
> `E0516 01:53:16.520118 140269701428992 xds_k8s_testcase.py:320] Rpcs did not go to expected servers before timeout 600`
>
> `AssertionError: 'psm-grpc-server-6df55f5cf9-7rh5b' not found in ['psm-grpc-server-6df55f5cf9-lk6cc', 'psm-grpc-server-6df55f5cf9-ln29c', 'psm-grpc-server-6df55f5cf9-p46fc', 'psm-grpc-server-6df55f5cf9-q6frs'] : Unexpected server psm-grpc-server-6df55f5cf9-7rh5b received RPCs`

**Comment by eugeneo** [comment]
@sergiitk I merged C++ changes

File: tools/run_tests/xds_k8s_test_driver/tests/outlier_detection_test.py [hunk 1]
```diff
@@ -55,10 +55,19 @@ def setUpClass(cls):
...
-            # TODO(mlumish): Once rpc-behavior supported by a language, make the
-            #                override version-conditional.
-            cls.server_image = xds_k8s_flags.SERVER_IMAGE_CANONICAL.value
+        # gRPC Java implemented server "error-code-" rpc-behavior in v1.47.x.
+        if cls.lang_spec.client_lang == _Lang.JAVA:
+            return
+
+        # gRPC CPP implemented server "error-code-" rpc-behavior in v1.55.x.
+        if (cls.lang_spec.client_lang == _Lang.CPP and
+                cls.lang_spec.version_gte('v1.55.x')):
```
Comment (eugeneo) [inline tools/run_tests/xds_k8s_test_driver/tests/outlier_detection_test.py:64]: I just merged C++ changes, they are not getting into 1.56

```suggestion
                cls.lang_spec.version_gte('v1.57.x')):
```

**Comment by sergiitk** [comment]
CPP test run:
- [ ] [grpc/core/master/linux/grpc_xds_k8s_lb](https://source.cloud.google.com/results/invocations/c52b4c3b-f292-41c9-a204-74c4994029ac)

**Pull request merged**